Apps/WebClient/src/NewClientApp/src/components/private/timeline/entry/Covid19TestResultTimelineComponent.vue

```diff
@@ -1,3 +1,219 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faDownload } from "@fortawesome/free-solid-svg-icons";
+import { saveAs } from "file-saver";
+import { computed, ref } from "vue";
+
+import DisplayFieldComponent from "@/components/common/DisplayFieldComponent.vue";
+import HgButtonComponent from "@/components/common/HgButtonComponent.vue";
+import MessageModalComponent from "@/components/common/MessageModalComponent.vue";
+import Covid19TestResultDescriptionComponent from "@/components/private/timeline/entry/Covid19TestResultDescriptionComponent.vue";
+import TimelineEntryComponent from "@/components/private/timeline/TimelineEntryComponent.vue";
+import { EntryType, entryTypeMap } from "@/constants/entryType";
+import { ErrorSourceType, ErrorType } from "@/constants/errorType";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import { DateWrapper } from "@/models/dateWrapper";
+import { ResultError } from "@/models/errors";
+import Covid19TestResultTimelineEntry from "@/models/timeline/covid19TestResultTimelineEntry";
+import { ILaboratoryService, ILogger } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+import { useTimelineStore } from "@/stores/timeline";
+import SnowPlow from "@/utility/snowPlow";
+
+library.add(faDownload);
+
+interface Props {
+    hdid: string;
+    entry: Covid19TestResultTimelineEntry;
+    index: number;
+    datekey: string;
+    isMobileDetails?: boolean;
+    commentsAreEnabled?: boolean;
+}
+const props = withDefaults(defineProps<Props>(), {
+    isMobileDetails: false,
+    commentsAreEnabled: false,
+});
+
+const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+const laboratoryService = container.get<ILaboratoryService>(
+    SERVICE_IDENTIFIER.LaboratoryService
+);
+const errorStore = useErrorStore();
+const timelineStore = useTimelineStore();
+
+const isLoadingDocument = ref(false);
+
+const sensitiveDocumentModal =
+    ref<InstanceType<typeof MessageModalComponent>>();
+
+const cols = computed(() => timelineStore.columnCount);
+const entryIcon = computed(
+    () => entryTypeMap.get(EntryType.Covid19TestResult)?.icon
+);
```

Comment:
This looks like we could move it into the TimelineEntryComponent, and derive this from entry being passed into the component.

the timlineEntry model has the type property which point to the EntryType value of the model. This could thus be centralized to the component.

I would argue that we leave the entry-icon property as an override though.